agent/component/worker/handler.go

```diff
@@ -44,26 +40,23 @@ func (w *Worker) PostRunResourceHandler(bCtx *env.BubblyContext, m *nats.Msg) er
 		Msg("processing request to run resource")
 
 	res, err := w.getRunResource(bCtx, wr.Name)
-
 	if err != nil {
-		return fmt.Errorf("interval worker failed to get resource: %w", err)
+		return nil, fmt.Errorf("interval worker failed to get resource: %w", err)
 	}
 
 	// parse the resource and add it to the worker's pool
 	err = w.ResourceWorker.ParseResource(bCtx, res, wr.RemoteInput)
-
 	if err != nil {
-		return fmt.Errorf("interval worker failed to parse resource: %w", err)
+		return nil, fmt.Errorf("interval worker failed to parse resource: %w", err)
 	}
 
 	// TODO: Support Interval Runs
 	err = w.ResourceWorker.RunOneOffRuns(bCtx)
-
 	if err != nil {
-		return fmt.Errorf("interval worker failure: %w", err)
+		return nil, fmt.Errorf("interval worker failure: %w", err)
 	}
 
-	return nil
+	return nil, nil
```

Comment:
I find this somewhat disturbing, but I can't explain why 😅

Reply:
I think this is fine. It's only odd until you realize it is the way is due to `SubscriptionHandlerFn `, as Jacob said.